lib/std/uuid.zig

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const rand = @import("std.zig").rand;
+
+const Self = @This();
+id: u128,
+
+/// Creates a brand new UUID v4 object given a random number generator.
+pub fn newv4(gen: *rand.Random) Self {
+    //This sets the 4 bits at the 48 bit to 0
+    const flip: u128 = 0b1111 << 48;
+
+    //We then set the 4 bits at 48 to 0x4 (0b0100)
+    return Self{ .id = ((gen.int(u128) & ~flip) | (0x4 << 48)) };
+}
+
+/// Creates a string from the ID.
+/// Format: xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx
+pub fn format(value: Self, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) !void
+    var buf: [36]u8 = [_]u8{0} ** 36; //Set to 0
+
+    const chars = "0123456789ABCDEF";
+
+    //Pre-set known values
+    buf[8] = '-';
+    buf[13] = '-';
+    buf[14] = '4';
+    buf[18] = '-';
+    buf[23] = '-';
+
+    //Generate the string
+    var i: usize = 0;
+    var byte: usize = 0;
```

Comment:
You could have this as a `u7` to avoid the `@intCast` below. 

Could also have a `u5` (or perhaps a `u6`, as the final `byte += 1` will take it over a `u5` I think?) but then for the right shift you'd need `>> (@as(u7, byte) * 4)`